src/com/linkedin/parseq/TaskDescriptorFactory.java

```diff
@@ -0,0 +1,69 @@
+package com.linkedin.parseq;
+
+import java.util.Iterator;
+import java.util.ServiceConfigurationError;
+import java.util.ServiceLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+class TaskDescriptorFactory {
+
+  static final Logger LOGGER = LoggerFactory.getLogger(TaskDescriptorFactory.class);
+
+  private static ServiceLoader<TaskDescriptor> _loader;
+  private static TaskDescriptor _identifier;
+
+  static {
+    try {
+      _loader = ServiceLoader.load(TaskDescriptor.class);
+    } catch (Throwable e) {
+      LOGGER.error("Unable to load providers of TaskDescriptor", e.getMessage());
```

Comment:
Will there be error logged if there is no TaskDescriptor provided i.e. the current case?